nanopub/profile.py

```diff
@@ -20,46 +14,86 @@ class ProfileError(RuntimeError):
     pass
 
 
-def get_orcid_id():
-    orcid_id = get_profile()[ORCID_ID]
-    if not orcid_id:
-        raise ProfileError('Your profile was not setup yet or not setup correctly. To setup '
-                           'your profile see instructions in Readme.md')
-    return orcid_id
+class Profile:
+    """Represents a user profile.
+
+    Attributes:
+        orcid_id: The user's ORCID
+        name: The user's name
+        public_key: Path to the user's public key
+        private_key: Path to the user's private key
+        nanopub_uri: URI of the user's profile nanopub
+    """
+    def __init__(
+            self,
+            orcid_id: str, name: str,
+            public_key: Path, private_key: Path,
+            nanopub_uri: Optional[str] = None
+            ) -> None:
+        """Create a Profile."""
+        self.orcid_id = orcid_id
+        self.name = name
+        self.public_key = public_key
+        self.private_key = private_key
+        self.nanopub_uri = nanopub_uri
+
+    @classmethod
+    def _yatiml_savorize(cls, node: yatiml.Node) -> None:
+        node.rename_attribute('profile_nanopub', 'nanopub_uri')
+
+    @classmethod
+    def _yatiml_sweeten(cls, node: yatiml.Node) -> None:
+        node.remove_attributes_with_default_values(cls)
+        node.rename_attribute('nanopub_uri', 'profile_nanopub')
+
+
+_load_profile = yatiml.load_function(Profile)
+
 
+_dump_profile = yatiml.dump_function(Profile)
 
-def get_public_key():
-    filepath = get_profile()[PUBLIC_KEY_FILEPATH]
-    with open(filepath, 'r') as f:
-        public_key = f.read()
-    if not public_key:
-        raise ProfileError('Your profile was not setup yet or not setup correctly. To setup '
-                           'your profile see instructions in Readme.md')
-    return public_key
+
+def get_orcid_id() -> str:
+    """Returns the user's ORCID."""
+    return get_profile().orcid_id
+
+
+def get_public_key() -> str:
+    """Returns the user's public key."""
+    return get_profile().public_key
```

Comment:
This doesn't return the key itself anymore, but the path to it right? And there's apparently no test that breaks :O

Reply:
Oh you're right, I overlooked that, will fix. It passes the tests as-is, I ran them locally before committing of course. So this is a coverage issue, not a CI issue.

Reply:
Fixed, although I'm checking for a missing file now rather than an empty file. I'm not sure what the expected error modes are, but at least you get a missing file if the profile has not been set up.

Reply:
Ha, here's a good argument for increasing test coverage :)